crates/rune-modules/src/base64.rs

```diff
@@ -0,0 +1,37 @@
+use base64::prelude::*;
+use rune::{ContextError, Module};
+
+/// Construct the `base64` module.
+pub fn module(_stdio: bool) -> Result<Module, ContextError> {
+    let mut module = Module::with_crate("base64")?;
+    module.item_mut().docs([
+        "Base64 encoding and decoding",
+        "",
+        "This module provides functions to encode & decode base64 data.",
+    ])?;
+
+    module.function_meta(decode)?;
+    module.function_meta(encode)?;
+    Ok(module)
+}
+
+/// Decode a base64 String into data
+///
+/// ```rune
+/// //assert_eq!(base64::decode("+uwgVQA=")?, b"\xFA\xEC\x20\x55\0");
+/// assert!(base64::decode("+uwgVQA=").is_ok());
+/// ```
+#[rune::function]
+fn decode(inp: &str) -> Result<Vec<u8>, String> {
+    BASE64_STANDARD.decode(inp).map_err(|e| e.to_string())
```

Comment:
Please define a wrapper around the base64 error typ8e, else the error type exported to Rune scripts will be strings rather than an appropriate error type.

The error type primarily needs to implement the `STRING_DISPLAY` protocol. You can see [an example here](https://github.com/rune-rs/rune/blob/52c52f5d93c2336aa4434cd713c619384b70a2c1/crates/rune-modules/src/http.rs#L96).

Reply:
Done, Thanks for the hint.